onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -170,6 +193,47 @@ bool AreRangesEqual(const Range& lhs, const Range& rhs) {
          std::equal(lhs.begin(), lhs.end(), rhs.begin());
 }
 
+
+bool AreEqual(const ONNX_NAMESPACE::TensorProto* lhs, const ONNX_NAMESPACE::TensorProto* rhs)
+{
+  if((lhs == nullptr) || (rhs == nullptr))
+    return ((lhs == nullptr) == (rhs == nullptr));
+  
+  if(lhs->data_type() != rhs->data_type())
+    return false;
+  
+  if(lhs->has_raw_data() && rhs->has_raw_data()) 
+    return AreRangesEqual(lhs->raw_data(), rhs->raw_data());
+
+  switch(lhs->data_type()) {
+    case ONNX_NAMESPACE::TensorProto_DataType_INT8:
+    case ONNX_NAMESPACE::TensorProto_DataType_INT16:
+    case ONNX_NAMESPACE::TensorProto_DataType_INT32:
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT16:
+    case ONNX_NAMESPACE::TensorProto_DataType_BOOL:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+      return AreRangesEqual(lhs->int32_data(), rhs->int32_data());
+    
+    case ONNX_NAMESPACE::TensorProto_DataType_STRING:
+      return AreRangesEqual(lhs->string_data(), rhs->string_data());
+      
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64:
+      return AreRangesEqual(lhs->int64_data(), rhs->int64_data());
+      
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+    case ONNX_NAMESPACE::TensorProto_DataType_COMPLEX64:
+      return AreRangesEqual(lhs->float_data(), rhs->float_data());
+      
+    case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE:
+    case ONNX_NAMESPACE::TensorProto_DataType_COMPLEX128:
+      return AreRangesEqual(lhs->double_data(), rhs->double_data());
+      
+    default:
+      return AreRangesEqual(lhs->raw_data(), rhs->raw_data());
```

Comment:
I guess we should return false for safety here (in case another datatype and the corresponding `_data` field is added at some point).